app/Http/Controllers/MyGalleryController.php

```diff
@@ -40,24 +36,12 @@ public function create(Request $request): Response|RedirectResponse
         /** @var User $user */
         $user = $request->user();
 
-        $collections = $user
-            ->collections()
-            ->withUserNftsCount($user)
-            ->orderBy('id')
-            ->paginate((int) config('dashbrd.gallery.pagination.collections_per_page'));
-
-        $collections->withPath(route('my-galleries.collections'));
-
-        $nftsPerPage = (int) config('dashbrd.gallery.pagination.nfts_per_page');
-
-        $nfts = Nft::paginatedCollectionNfts($collections, $user, $nftsPerPage);
-
         return Inertia::render('Galleries/MyGalleries/Create', [
             'title' => trans('metatags.my_galleries.create.title'),
-            'nfts' => GalleryNftData::collection($nfts),
-            'collections' => new GalleryCollectionsData(GalleryCollectionData::collection($collections)),
-            'nftsPerPage' => $nftsPerPage,
+            'nftsPerPage' => (int) config('dashbrd.gallery.pagination.nfts_per_page'),
+            'collectionsPerPage' => (int) config('dashbrd.gallery.pagination.collections_per_page'),
             'nftLimit' => config('dashbrd.gallery.nft_limit'),
+            'nftCount' => $user->nfts->count(),
```

Comment:
is there a reason you don't use query builder to get the count? `$user->nfts()->count()`. your way it loads all NFTs that user owns in a memory

Reply:
Ah, $hit, I didn't pay too much attention into this line as I copied it from the method [above](https://github.com/ArdentHQ/dashbrd/pull/370/files#diff-cc0cc28e719e766a0e67369768c8a629c7b2201759058408ed67d22ca0ba1c44R30), fixed all instances.